libkodimote/kodiconnection.cpp

```diff
@@ -259,10 +247,24 @@ void KodiConnectionPrivate::slotConnected()
 {
     koDebug(XDAREA_CONNECTION) << "Connected to remote host. Asking for version...";
 
-    m_versionRequestId = m_commandId++;
-    Command cmd(m_versionRequestId, "JSONRPC.Version");
-    m_commandQueue.prepend(cmd);
-    sendNextCommand2();
+    m_versionRequestId = sendCommand("JSONRPC.Version", QVariant(), this, "versionReceived");
+}
+
+void KodiConnectionPrivate::versionReceived(const QVariantMap &rsp)
+{
+    // If the remote xbmc has the version field not compiled in, assume its the latest known one
+    koDebug(XDAREA_CONNECTION) << "Parsing XBMC version:" << rsp.value("result").toMap().value("version");
+    m_kodiVersionMajor = rsp.value("result").toMap().value("version").toMap().value("major").toInt();
+    if(m_kodiVersionMajor < 6) {
+        qDebug() << "WARNING! XBMC is too old or version field not valid! Some features might not work";
+        m_connectionError = tr("This version of Kodimote is designed to work with XBMC Frodo (v12.0). It seems you have connected to an older version of XMBC. Please upgrade XBMC in order to use Kodimote.");
+    } else {
+        m_connected = true;
+        m_connectionError.clear();
+    }
+    m_connecting = false;
+    emit m_notifier->connectionChanged();
+    m_host->setPersistent(true);
```

Comment:
Behavior is the same as current master: https://github.com/mzanetti/kodimote/blob/21b60c0544702498acbe1942003ed5cfe25fece5/libkodimote/kodiconnection.cpp#L407

This code is only ran when:
1. The socket is opened/connected (as it is called from slotConnected)
2. The HTTP calls work (as the version command runs over HTTP)
So this is already done when you are successfully connected.